sys/Makefile.include

```diff
@@ -86,3 +86,12 @@ endif
 ifneq (native,$(BOARD))
   INCLUDES += -I$(RIOTBASE)/sys/libc/include
 endif
+
+ifneq (,$(filter shell_commands,$(USEMODULE)))
+ifneq (,$(filter cortexm_common,$(USEMODULE)))
+  LINKFLAGS += $(RIOTBASE)/sys/shell/commands/shell_commands_cortexm.ld
+else
+  LINKFLAGS += $(RIOTBASE)/sys/shell/commands/shell_commands.ld
+endif
+  UNDEF += -Wl,--whole-archive $(BINDIR)/shell_commands.a  -Wl,--no-whole-archive
```

Comment:
Mmm, interesting. I was wondering how you avoided the linker discarding the symbols, which is the reason why #9105 is still unmerged: it needs either #8711 (ld -r) or "--whole-archive" (which I think is the preferred option in conjuction with #10195 )

I see that you avoided much of the problems that have been holding #8711 by limiting the "--whole-archive" to just the shell module.

@cladmi , @gebart Wouldn't this be a good approach "unlock" #8711, or its equivalent with "--whole-archive"? As far as I understand, most of the problems come from a minority of files which define things like interrupt handlers. The "--whole-archive" could be enabled only for select modules (or, if most things work, enabled by default and disabled for select modules.)

Reply:
For #8711 I was thinking about rebasing/cleaning it, changing it to `whole-archive` and going for trying to enable per architecture. It managed to go far enough so that it is only a few PRs away for being able to enable some boards.

Not to say it should be supported only on some boards, but simplify the review.

We now have a server to build the two versions and compare the elf files, just need to take the time to analyze and solve all the issues.

Reply:
> trying to enable per architecture.

Agree. I insist, if there is some problematic module (if I recall correcyly we were having _too few_ symbols discarded sometimes?) we can, as a quirk, disable it just for that module.